packages/cli/e2e-test/helpers.js

```diff
@@ -117,12 +117,29 @@ async function waitForPageWithText(
     }
   }
 
+  // The page may not be fully loaded and hence we need to retry.
+  const maxAttemptsToSearchText = 3;
+  let attemptsToSearchText = 0;
   const escapedText = text.replace(/"/g, '\\"');
-  browser.assert.evaluate(
-    `Array.from(document.querySelectorAll("*")).some(el => el.textContent === "${escapedText}")`,
-    true,
-    `expected to find text ${text}`,
-  );
+  for (;;) {
+    try {
+      browser.assert.evaluate(
+        `Array.from(document.querySelectorAll("*")).some(el => el.textContent === "${escapedText}")`,
+        true,
+        `expected to find text ${text}`,
+      );
+      break;
+    } catch (error) {
+      if (error instanceof browser.assert.AssertionError) {
```

Comment:
Can prolly do this for any error? Less code :grin:

Reply:
Yeah, done!